dbms/src/Storages/Page/V3/LogFile/LogWriter.cpp

```diff
@@ -92,40 +99,32 @@ void LogWriter::close()
 void LogWriter::addRecord(ReadBuffer & payload, const size_t payload_size, const WriteLimiterPtr & write_limiter)
```

Comment:
https://github.com/pingcap/tiflash/blob/3119fa401635635cfe7e7369f95d4250fcfeb63b/dbms/src/Storages/Page/V3/WALStore.cpp#L201-L206

We should add a bool param `background` and set `write_limiter`, `background=true` inside `WALStore::saveSnapshot`. Or it won't be limited and counted as the background write flow.

Reply:
Fixed. By the way, it seems we may call `WALStore::apply` in a background thread but `addRecord` is always called with `background=true`. https://github.com/pingcap/tiflash/blob/79b36c3bf30dc2604b66971a54616512617e9676/dbms/src/Storages/Page/V3/WALStore.cpp#L105
Do we need to fix this? `BlobFile` also has similar problem.

Reply:
May be we can check background like `RateLimiter` using `is_background_thread` in `PageUtil::writeFile` and avoid pass `background` around?

Reply:
`background` maybe a misleading name. I want to check the write flow that is caused by compaction (`saveSnapshot` as log_file compaction, `full gc` as blob_file compaction) and compare it to the total write flow.

Reply:
the thread local variable `is_background_thread` may be some write caused by compaction in the DeltaTree level, that is not what I want.

Reply:
`WALStore::apply(const PageEntriesEdit & edit, const WriteLimiterPtr & write_limiter)` is a common method that will be used for both write from out of the `PageStorage` level and write from the compaction inside `PageStorage` level.